### PR TITLE
fix broken links

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #293, #381, #402

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by crodriguezvega** [comment]
There are still 3 links that are broken, but I need some advice for 2 of them:

- https://ibc.cosmos.network - this will be fixed when the docs website is live, which I hope will happen soon.
- https://media.githubusercontent.com/media/cosmos/ics/master/spec/ics-004-channel-and-packet-semantics/packet-state-machine.png (used [here](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps.md#packet-flow-handling)).
- https://github.com/cosmos/ics/blob/master/ibc/1_IBC_TERMINOLOGY.md (used [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/spec/01_concepts.md)).

I also updated the links that were broken in `go` and `proto` files. I didn't touch the generated `pb.go` files. Let me know if I should do something else regarding this.

**Comment by colin-axner** [comment]
>     * https://ibc.cosmos.network - this will be fixed when the docs website is live, which I hope will happen soon.

This is totally fine, let's ignore for now
 
>     * https://media.githubusercontent.com/media/cosmos/ics/master/spec/ics-004-channel-and-packet-semantics/packet-state-machine.png (used [here](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps.md#packet-flow-handling)).

We can use [this](https://github.com/cosmos/ibc/blob/old/spec/ics-004-channel-and-packet-semantics/channel-state-machine.png)

>     * https://github.com/cosmos/ics/blob/master/ibc/1_IBC_TERMINOLOGY.md (used [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/spec/01_concepts.md)).

I'm in favor of removing. The document was outdated and deprecated. It can be found [here](https://github.com/cosmos/ibc/blob/old/ibc/1_IBC_TERMINOLOGY.md)

> I also updated the links that were broken in `go` and `proto` files. I didn't touch the generated `pb.go` files. Let me know if I should do something else regarding this.

If you changed links in .proto files, you need to run `make proto-all` to update the .pb.go and proto docs file

**Review by colin-axner — APPROVED** [review]
This is very exciting! Broken links have been an issue since migrating we migrated to ibc-go and the spec repository underwent significant refactoring

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -5,4 +5,6 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - uses: gaurav-nelson/github-action-markdown-link-check@v1
\ No newline at end of file
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        config-file: '.github/workflows/link-check-config.json'
```
Comment (damiannolan) [inline .github/workflows/link-check.yml:10]: Nice! This did the trick 👍

**Review by damiannolan — APPROVED** [review]

**Review by seantking — APPROVED** [review]

**Review by seantking — APPROVED** [review]

**Pull request merged**